src/plugin/time/resource/RealTimeProvider.cpp

```diff
@@ -0,0 +1,15 @@
+#include "RealTimeProvider.hpp"
+
+namespace ES::Plugin::Time::Resource {
+RealTimeProvider::RealTimeProvider() { _lastTime = std::chrono::high_resolution_clock::now(); }
+
+float RealTimeProvider::GetElapsedTime()
+{
+    auto currentTime = std::chrono::high_resolution_clock::now();
+    auto elapsedTime = std::chrono::duration<float>(currentTime - _lastTime).count();
+
+    return elapsedTime;
```

Comment:
This will get elapsed time since the last update but there will be a moment between the last call of `GetElapsedTime` and next update which will not be comptabilized. Moreover, each call of this function will make some calculus rather than just getter a precalculated value. You should just get a value in `GetElapsedTime`. `Update` method should update time since the last call of `Update`.

Reply:
fixed :)

Reply:
please have a look at the new implementation

Reply:
👍🏼